File: Python/728. Self Dividing Numbers.py
# Time: O(nm) 
class Solution:
    def judge(self, x):
        ss = str(x)
        for s in ss:
            if s=='0' or x % int(s) != 0:
                return False
        return True

    def selfDividingNumbers(self, left, right):
        """
        :type left: int
        :type right: int
        :rtype: List[int]
        """
        ans = list()
        for i in range(left, right+1):
            if self.judge(i):
                ans.append(i)
        return ans

File: C++/728. Self Dividing Numbers.cpp
// Time: O(nlog(r))
class Solution {
public:
    vector<int> selfDividingNumbers(int left, int right) {
        vector<int>ans;
        for(int i=left;i<=right;i++)
        {
            int tmp = i,flag = 1;
            while(tmp)
            {
                int k = tmp%10;
                if(k==0 || i%k!=0)
                {
                    flag = 0;
                    break;
                }
                tmp /= 10;
            }
            if(flag)
                ans.push_back(i);
        }
        return ans;
    }
};